viewer/src/main/java/nl/b3p/viewer/util/FeatureToJson.java

```diff
@@ -303,12 +349,24 @@ private JSONObject populateWithRelatedFeatures(JSONObject j,SimpleFeature featur
                 }
             }
             if (related_featuretypes.length()>0){
-                j.put("related_featuretypes",related_featuretypes);
+                if(ordered == null) {
+                    j.put("related_featuretypes",related_featuretypes);
+                } else {
+                    JSONObject relatedFeatures = new JSONObject();
+                    relatedFeatures.put("related_featuretypes", related_featuretypes);
+                    ordered.put(relatedFeatures);
+                }
             }
         }
         return j;
     }
 
+    private void concatArray(JSONArray arr1, JSONArray arr2) throws JSONException {
+        for (int i = 0; i < arr2.length(); i++) {

```

Comment:
probably there is a JSONArray#addAll() because I think there is a java collection backing the JSONArray, maybe depends which impl this is... see eg. https://docs.oracle.com/javaee/7/api/javax/json/JsonArray.html

Reply:
It uses an other implementation: http://www.docjar.com/docs/api/org/json/JSONArray.html which does not have an addAll/putAll method